pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -2999,3 +2999,113 @@ func TestFilterChainMatchEqual(t *testing.T) {
 		})
 	}
 }
+
+func TestOutboundListenerConfig_WithAutoAllocatedAddress(t *testing.T) {
+	const tcpPort = 79
+	services := []*model.Service{
+		{
+			CreationTime:             tnow.Add(1 * time.Second),
+			Hostname:                 host.Name("test1.com"),
+			DefaultAddress:           wildcardIPv4,
+			AutoAllocatedIPv4Address: "240.240.0.100",
+			Ports: model.PortList{
+				&model.Port{
+					Name:     "tcp",
+					Port:     tcpPort,
+					Protocol: protocol.TCP,
```

Comment:
Please also add case for http port

Reply:
HTTP uses `hosts` to multiplex traffic, it doesn't have problem with bind to `0.0.0.0`; I propose covering HTTP case with another PR if it's required.